chia/full_node/full_node.py

```diff
@@ -494,6 +501,12 @@ async def _handle_one_transaction(self, entry: TransactionQueueEntry) -> None:
         try:
             inc_status, err = await self.add_transaction(entry.transaction, entry.spend_name, peer, entry.test)
             entry.done.set((inc_status, err))
+            # if the transaction is valid, it was added to the mempool and we
+            # don't need to keep it in _recent_txs anymore
+            try:
+                self._recent_txs.remove(entry.spend_name)
```

Comment:
Why are we not checking `inc_status, err` before we do this? the comment implies we only remove this item when `add_transaction` succeeds, no?

Reply:
oh uh. you're right. I saw the try-block and expected errors to throw